File: solutions/elixir/bob/1/lib/bob.ex
defmodule Bob do
  def hey(input) do
    cond do
      question?(input) && yelling?(input) -> "Calm down, I know what I'm doing!"
      question?(input) -> "Sure."
      yelling?(input) -> "Whoa, chill out!"
      blank?(input)-> "Fine. Be that way!"
      true -> "Whatever."
    end
  end

  defp question?(input), do: String.ends_with?(String.trim(input), "?")
  defp with_alpha?(input), do: String.upcase(input) != String.downcase(input)
  defp yelling?(input), do: with_alpha?(input) and input == String.upcase(input)
  defp blank?(input), do: String.trim(input) == ""
end

File: solutions/javascript/bob/1/bob.js
const isAQuestion = (message) => message.endsWith('?');
const isShouting  = (message) =>
  message.toUpperCase() === message && /[A-Z]/.test(message);
const isSilence   = (message) => message.replace(/\s+/g, '') === '';

export function hey(message) {
  if (isSilence(message)) return 'Fine. Be that way!';
  if (isShouting(message)) {
    if (isAQuestion(message)) return "Calm down, I know what I'm doing!";
    return 'Whoa, chill out!';
  }
  if (isAQuestion(message.trim())) return 'Sure.';
  return 'Whatever.';
};